File: src/main/java/com/tecnogeek/comprameya/service/RecursoService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.tecnogeek.comprameya.service;

import com.tecnogeek.comprameya.entidad.Recurso;
import com.tecnogeek.comprameya.repositories.BaseRepository;
import com.tecnogeek.comprameya.repositories.RecursoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 *
 * @author alexander
 */

@Service
public class RecursoService extends BaseService<Recurso, Long>{

    @Autowired
    RecursoRepository recursoRepository;
    
    @Override
    public BaseRepository<Recurso, Long> getRepository() {
        return recursoRepository;
    }
    
    public Iterable<Recurso> getRecursos(Long idPublicacion, int page, int recursosByPage){
        return recursoRepository.getRecursos(idPublicacion, page, recursosByPage);
    }
    
}

File: cmpy/src/main/webapp/resources/js/anuncios/recursoService.js
modulo_anuncios.service('recursoService',
        ['$http',
            '$log',
            function ($http, $log) {

                this.getRecursos = function (idPublicacion, page) {
                    var req = "/recurso/getRecursos/" + idPublicacion + "/" + page;
                    return $http.get(req)
                            .success(function (response)
                            {
                                return response;
                            });
                };

            }]);